common/core/desktop/src/kmx/kmx_xstring.cpp

```diff
@@ -119,33 +119,57 @@ km_kbp_cp *km::kbp::kmx::u16tok(km_kbp_cp *p, km_kbp_cp ch, km_kbp_cp **ctx) {
 
 PKMX_WCHAR km::kbp::kmx::incxstr(PKMX_WCHAR p)
 {
-  if(*p == 0) return p;
-  if(*p != UC_SENTINEL)
+  int deltaptr;			// how many bytes to jump over 
+
+  if (*p == 0) return p;
+  if (*p != UC_SENTINEL)
   {
-    if(*p >= 0xD800 && *p <= 0xDBFF && *(p+1) >= 0xDC00 && *(p+1) <= 0xDFFF) return p+2;
-    return p+1;
+    if (*p >= 0xD800 && *p <= 0xDBFF && *(p + 1) >= 0xDC00 && *(p + 1) <= 0xDFFF) return p + 2;
+    return p + 1;
   }
-
-  p+=2;
-  switch(*(p-1))
+  else
```

Comment:
The `else {}` is not needed here because the previous `if` block always `return`s. So we can move the code contained in the else block out to the parent level.

Reply:
... removed :-)